test/test_click_api_consistency.py

```diff
@@ -47,10 +49,27 @@ def test_api_consistency(self):
         for cli_call, api_call in zip(click_commands, api_functions):
             cli_spec = inspect.getfullargspec(cli_call.callback)
             api_spec = inspect.getfullargspec(api_call)
-            self.assertEqual(api_spec.args, cli_spec.args)
-            self.assertEqual(api_spec.annotations, cli_spec.annotations)
-            self.assertEqual(api_spec.args, self._param_names_of_click_call(cli_call))
+
+            exaslct_utils.multiassert([lambda: self.assertEqual(api_spec.args, cli_spec.args),
+                                       lambda: self.assertEqual(api_spec.annotations, cli_spec.annotations),
+                                       lambda: self.assertEqual(api_spec.args,
+                                                                self._param_names_of_click_call(cli_call))], self)
+
+    def test_api_default_values(self):
+        """
+        Validate that the default values for all commands match!
+        """
+
+        # Get all click commands in module exasol_script_languages_container_tool.cli.commands
+        click_commands = [c[1] for c in inspect.getmembers(commands, is_click_command)]
+        # Get all functions in module exasol_script_languages_container_tool.lib.api
+        api_functions = [f[1] for f in inspect.getmembers(api, inspect.isfunction)]
+
+        # Now iterate over the list and compare consistency
+        for cli_call, api_call in zip(click_commands, api_functions):
+            api_spec = inspect.getfullargspec(api_call)
             cli_defaults = self._defaults_of_click_call(cli_call)
+
             self.assertEqual(len(cli_defaults), len(api_spec.defaults))
```

Comment:
this is in theory also a multi assert and actually, one which can hide something

Reply:
the fail below as well

Reply:
maybe, it it possible to replace this with parameterized tests

Reply:
I was investigating how to use parametrized test with Python's unittest framework, but it would depend on another 3rd-party package. So I didn't pursue this...

Reply:
> this is in theory also a multi assert and actually, one which can hide something

If the length don't match it doesn't make sense to continue the test, as it would compare default values of different arguments.

Reply:
but you do the comparison for multiple commands, and you might want to know, which of the commands have the problem and don't stop for the first one.